schema-serializer/src/main/java/io/confluent/kafka/serializers/migration/CompositeDeserializer.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2024 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+
+package io.confluent.kafka.serializers.migration;
+
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDe;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CompositeDeserializer implements Deserializer<Object> {
+
+  protected Logger log = LoggerFactory.getLogger(CompositeDeserializer.class);
+
+  protected static final byte MAGIC_BYTE = 0x0;
+
+  private boolean isKey;
+  private Deserializer<?> oldDeserializer;
+  private Deserializer<?> confluentDeserializer;
+  private SchemaRegistryClient schemaRegistryClient;
+
+  /**
+   * Constructor used by Kafka consumer.
+   */
+  public CompositeDeserializer() {
+  }
+
+  @Override
+  public void configure(Map<String, ?> configs, boolean isKey) {
+    configure(new CompositeDeserializerConfig(configs), isKey);
+  }
+
+  protected void configure(CompositeDeserializerConfig config, boolean isKey) {
+    this.isKey = isKey;
+    Map<String, Object> originals = config.originals();
+    this.oldDeserializer = config.getConfiguredInstance(
+        CompositeDeserializerConfig.OLD_DESERIALIZER, Deserializer.class);
+    this.oldDeserializer.configure(originals, isKey);
+    this.confluentDeserializer = config.getConfiguredInstance(
+        CompositeDeserializerConfig.CONFLUENT_DESERIALIZER, Deserializer.class);
+    this.confluentDeserializer.configure(originals, isKey);
+    this.schemaRegistryClient = getSchemaRegistryClient();
+  }
+
+  @Override
+  public Object deserialize(String topic, byte[] bytes) {
+    return deserialize(topic, null, bytes);
+  }
+
+  @Override
+  public Object deserialize(String topic, Headers headers, byte[] bytes) {
+    if (bytes == null) {
+      return null;
+    }
+
+    // We assume TopicNameStrategy
```

Comment:
why the limitation?

Reply:
This was for simplicity.  I can take a look at using the subject name strategy